### ISSUE TITLE
[Accessibility]: VaTable needs focusable caption element from table-title prop

### ISSUE THREAD

**Issue by OptionSelect** [department-of-veterans-affairs/vets-design-system-documentation#5099]
### Point of contact

Brandyn Sullins(@OptionSelect), Heather Rienks, or Joseph Lee(@josephlee)

### Defect Severity

2, Serious. Should be fixed in 1-2 sprints post-launch.

### Severity rationale

Table works as is with pagination but best practices indicate that focus should be used in this instance.

https://design.va.gov/accessibility/focus-management#the-focus-should-be-intentionally-set-to-the-appropriate-element-when-a-user-action-requires-a-change-of-context-or-location

https://www.w3.org/WAI/tutorials/tables/caption-summary/#summaries-for-more-complex-tables

See thread regarding convo: https://dsva.slack.com/archives/C8E985R32/p1761263820369879

### URL where issue occurs

https://staging.va.gov/manage-va-debt/summary/copay-balances/eea5e722-ced7-4812-a5e9-d1d32472f986

### Component or pattern (if applicable)

Va-Table

### Environment details

Chrome, Combined Debt portal signed in staging user 81
https://staging.va.gov/manage-va-debt/summary/copay-balances/ will precede any of the copay's detail pages on which the table lies.

### Steps to reproduce

Log into staging user 81
Navigate to https://staging.va.gov/manage-va-debt/summary/
Click copays link
Click any Review details link for a copay and scroll down to see the table

### Expected behavior / Acceptance criteria

Table title is focusable per guidelines. We are currently implementing pagination on this table which will re-render and we'd like to call focus to the new caption that will display "Showing X-Y of Z charges"

### Related issues or systemic notes

Pagination still in development. Jami's request is at the end of the thread. https://dsva.slack.com/archives/C8E985R32/p1761763724512989?thread_ts=1761263820.369879&cid=C8E985R32

### Type of issue

- [x] Axe-core
- [x] Cognitive | user flows & navigating
- [ ] Color, typography, & visual elements
- [x] Components and pattern standards
- [ ] Content organization
- [ ] Content style guide
- [x] Focus
- [ ] Headings
- [ ] Keyboard
- [ ] Link and button labels
- [ ] Markup and metadata
- [ ] Non-text content and media
- [ ] Overall user experience
- [ ] Screenreader | assistive tech & device support
- [ ] Semantic HTML
- [ ] URL standards
- [ ] Zoom or Magnification
- [ ] Something else

**Comment by OptionSelect** [comment]
FWIW we hope this is resolved ASAP, its not necessarily launch blocking but does hold up a fair bit of our work.
CC: @josephrlee @HeatherRienks

**Comment by HeatherRienks** [comment]
@mgleo @Marcela-FFTC
